sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/models/KeyVaultKeyIdentifier.java

```diff
@@ -16,11 +16,46 @@
 public final class KeyVaultKeyIdentifier {
     private final String keyId, vaultUrl, name, version;
 
-    private KeyVaultKeyIdentifier(String keyId, String vaultUrl, String name, String version) {
-        this.keyId = keyId;
-        this.vaultUrl = vaultUrl;
-        this.name = name;
-        this.version = version;
+    /**
+     * Create a new {@link KeyVaultKeyIdentifier} from a given key identifier.
+     *
+     * <p>Valid examples are:
+     *
+     * <ul>
+     *     <li>https://{key-vault-name}.vault.azure.net/keys/{key-name}</li>
+     *     <li>https://{key-vault-name}.vault.azure.net/keys/{key-name}/pending</li>
+     *     <li>https://{key-vault-name}.vault.azure.net/keys/{key-name}/{unique-version-id}</li>
+     *     <li>https://{key-vault-name}.vault.azure.net/deletedkeys/{deleted-key-name}</li>
+     * </ul>
+     *
+     * @param keyId The key identifier to extract information from.
+     *
+     * @throws IllegalArgumentException if the given identifier is {@code null} or an invalid Key Vault Key identifier.
+     */
+    public KeyVaultKeyIdentifier(String keyId) {
+        if (keyId == null) {
+            throw new IllegalArgumentException("keyId cannot be null");
+        }
+
+        try {
+            final URL url = new URL(keyId);
+            // We expect an identifier with either 2 or 3 path segments: collection + name [+ version]
+            final String[] pathSegments = url.getPath().split("/");
+
+            if ((pathSegments.length != 3 && pathSegments.length != 4) // More or less segments in the URI than expected.
+                || !"https".equals(url.getProtocol()) // Invalid protocol.
+                || (!"keys".equals(pathSegments[1]) && !"deletedkeys".equals(pathSegments[1])) // Invalid collection.
```

Comment:
Same comment here about asserting - rather, not needing to. IIRC, no other language does because it wasn't really important to. This is just a convenience/helper. There's no harm in, say, parsing a secret ID.

Reply:
Left [a comment](https://github.com/Azure/azure-sdk-for-java/pull/20357#discussion_r607427937) about this above.